src/tsickle.ts

```diff
@@ -2153,37 +2162,62 @@ function addClutzAliases(
       // https://github.com/angular/clutz/tree/master/src/test/java/com/google/javascript/clutz
-      const clutzSymbolName = 'module$contents$' + clutzModuleName + '_' + symbol.name;
-
+      // The first symbol name is that currently produced by clutz, and the second
+      // is what incremental clutz will produce.
+      const reexportData = [];
+      reexportData.push({
+        namespace: 'ಠ_ಠ.clutz',
+        clutzSymbolName: 'module$contents$' + clutzModuleName + '_' + symbol.name,
+        aliasedSymbolName: symbol.name
+      });
+      reexportData.push({
+        namespace: 'ಠ_ಠ.clutz.module$exports$' + clutzModuleName,
+        clutzSymbolName: symbol.name,
+        aliasedSymbolName: 'module$contents$' + clutzModuleName + '_' + symbol.name
+      });
       const {params, paramsWithContraint} = getGenericTypeParameters(symbol);
 
       if (symbol.flags & ts.SymbolFlags.Class) {
         // classes need special care to match clutz, which seperates class types into a
         // type for the static properties and a type for the instance properties
-        reexports.push(`type ${clutzSymbolName}${paramsWithContraint} = ${symbol.name}${params};`);
-        reexports.push(`const ${clutzSymbolName}: typeof ${symbol.name};`);
-        reexports.push(
-            `type ${clutzSymbolName}_Instance${paramsWithContraint} = ${symbol.name}${params};`);
-        reexports.push(`const ${clutzSymbolName}_Instance: typeof ${symbol.name};`);
-        continue;
+        reexportData.push({
+          namespace: 'ಠ_ಠ.clutz',
+          clutzSymbolName: 'module$contents$' + clutzModuleName + '_' + symbol.name + '_Instance',
+          aliasedSymbolName: symbol.name
+        });
+        reexportData.push({
+          namespace: 'ಠ_ಠ.clutz.module$exports$' + clutzModuleName,
+          clutzSymbolName: symbol.name + '_Instance',
+          aliasedSymbolName: 'module$contents$' + clutzModuleName + '_' + symbol.name
+        });
       }
 
-      if (symbol.flags & ts.SymbolFlags.Type) {
-        reexports.push(`type ${clutzSymbolName}${paramsWithContraint} = ${symbol.name}${params};`);
+      if (symbol.flags & ts.SymbolFlags.Type || symbol.flags & ts.SymbolFlags.Class) {
+        for (const {namespace, clutzSymbolName, aliasedSymbolName} of reexportData) {
+          addToMultiMap(
+              reexports, namespace,
+              `type ${clutzSymbolName}${paramsWithContraint} = ${aliasedSymbolName}${params};`);
+        }
       }
-      if (symbol.flags & ts.SymbolFlags.Value) {
-        reexports.push(`const ${clutzSymbolName}: typeof ${symbol.name};`);
+      if (symbol.flags & ts.SymbolFlags.Value || symbol.flags & ts.SymbolFlags.Class) {
+        for (const {namespace, clutzSymbolName, aliasedSymbolName} of reexportData) {
+          addToMultiMap(
+              reexports, namespace, `const ${clutzSymbolName}: typeof ${aliasedSymbolName};`);
+        }
       }
     }
   }
 
-  if (reexports.length) {
+  if (reexports.size) {
```

Comment:
I think .length was correct.

Reply:
`reexports` is a map, not a list.  I think your confusion come from `reexports` and `reexportsData` both existing.  I could rename them to `namespaceToReexportLineMultimap` and `reexportsDataList`, but that seems terrible.  Do you have any thoughts on improving the names?

Reply:
I might give the inner one the shorter name and the outer one the longer name.  Like maybe reexports for the inner one, and allReexports or reexportsBySymbol for the outer one.  ("Data" is usually a filler word so you can leave it out.)

Reply:
Renamed the outer one to `reexportsByNamespace` and the inner on to just `reexports`.